### DISCUSSION TITLE
Statuses and their meaning

### DISCUSSION THREAD

**Discussion by noamoss** [govflow/govflow#64] [Q&A] [answered]
This thread to elaborate on the meaning behind the current Service Request status types

**Comment by pwalsh** [comment]
@noamoss the current statuses for a service request https://github.com/govflow/govflow/blob/main/src/core/service-requests/models.ts#L5 are a hard coded, linear workflow for the lifecycle of a request. We always planned to make this customizable, but we have no need for that with existing users and so we haven't gone further in the design of such a feature. 

As for the meaning of each status (you asked me about Moved). 

The only *codified* meaning is that 

1. statuses are in an open or a closed state: https://github.com/govflow/govflow/blob/main/src/core/service-requests/models.ts#L16
2. "Inbox" is the default entry point for a request: https://github.com/govflow/govflow/blob/main/src/core/service-requests/models.ts#L95

The rest of the meaning to be derived from the current hard coded status list largely lies in how a jurisdiction will use it. 

However, The majority of statuses are so convention based that we can fairly safely rely on them have the same or a similar meaning for different users. In my reading, these ones are:

- inbox
- todo
- doing
- done
- invalid
- blocked (*possibly* - it could be that this seems obvious from technical project management but not to our users)

Then there are two other statuses we added to try to capture a specific meaning but we are not tracking if or how they are understood:

- moved (the issue has moved out of the scope of the request management process)
- junk (a type of invalid but specifically to label messages that are a form of spam)

@amirozer Do you agree? Anything to add?

**Reply by noamoss** [reply]
thanks @pwalsh .


1. about
>....moved (the issue has moved out of the scope of the request management process)

what does it mean in terms of the request management process? will this service request remain in this status forever? 



2.  How does the (current) hardcoded are related to the basic open/close ones? can I assume that 
"inbox", "todo", "doing" and "blocked" are considered/counted as "Open" requests
while "invalid", "moved" and "junk"k are treated as "Close" requests?

**Comment by pwalsh** [comment, marked as answer]
@noamoss to answer both your questions, the link I dropped above ( https://github.com/govflow/govflow/blob/main/src/core/service-requests/models.ts#L16 ) shows that the closed statuses are`['done', 'invalid', 'moved', 'junk']` and that means the others are all open statuses. Because `moved` is a closed status, yes, the request remains in this state, as with the other closed states.
